cmake/FindMKL.cmake

```diff
@@ -37,6 +37,9 @@ endif()
 set(INT_LIB "mkl_intel_lp64")
 set(THR_LIB "mkl_intel_thread")
 set(COR_LIB "mkl_core")
+set(DEF_LIB "mkl_def")
+set(AVX512_LIB "mkl_avx512")
+set(VMLAVX512_LIB "mkl_vml_avx512")
```

Comment:
Thank you for your contribution!
As far as I  know, if mkl_core is linked, mkl_avx512 has already linked. You do not need to explicitly link other libraries.
Here is an issue taking about your problem.
https://github.com/ikinsella/trefide/issues/2

Reply:
using LD_PRELOAD also works for me, thansk! This is really a strange error.
works
```
export LD_PRELOAD=/opt/intel/mkl/lib/intel64/libmkl_core.so:/opt/intel/mkl/lib/intel64/libmkl_def.so:/opt/intel/mkl/lib/intel64/libmkl_sequential.so
```
but when I link this three file in FindMKL.cmake  like
```
set(MKL_LIBRARIES 
  ${MKL_INTERFACE_LIBRARY} 
  ${MKL_THREADS_LIBRARY} 
  ${MKL_CORE_LIBRARY} 
  ${MKL_DEFINE_LIBRARY}
  ${MKL_SEQUENTIAL_LIBRARY}
  )
```
It will throw an error ```undefined symbol: mkl_serv_getenv```